### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by cua-cua**

### CODE REVIEW

File: encoder.c [hunk 1]
```diff
@@ -175,8 +151,13 @@ void update_encoder_readings(void)
 	uint16_t left_count;
 	uint16_t right_count;
 
+	float micrometers_per_count;
+	float wheels_separation;
+
 	left_count = read_encoder_left();
 	right_count = read_encoder_right();
+	micrometers_per_count = get_micrometers_per_count();
+	wheels_separation = get_wheels_separation();
```
Comment (Peque) [inline encoder.c]: `wheels_separation` is only used once. Maybe we could directly call the function there and save one variable declaration and a couple of lines?

File: command.c [hunk 2]
```diff
@@ -41,27 +42,37 @@ static void process_command(char *string)
...
+		set_control_constants(control);
+	} else if (starts_with(string, "set kp_angular_side ")) {
+		control.kp_angular_side =
+		    parse_float(string, RECEIVE_BUFFER_SIZE, 2);
+		set_control_constants(control);
+	} else if (starts_with(string, "set kp_angular_front ")) {
+		control.kp_angular_front =
+		    parse_float(string, RECEIVE_BUFFER_SIZE, 2);
+		set_control_constants(control);
+	} else
```
Comment (Peque) [inline command.c:66]: This function got even uglier... :joy:

But okay, we'll fix that later. :blush:

File: command.c [hunk 3]
```diff
@@ -33,15 +33,6 @@ static void process_command(char *string)
...
 		    parse_float(string, RECEIVE_BUFFER_SIZE, 2));
-	else if (starts_with(string, "set linear_acceleration "))
-		set_linear_acceleration(
-		    parse_float(string, RECEIVE_BUFFER_SIZE, 2));
-	else if (starts_with(string, "set linear_deceleration "))
-		set_linear_deceleration(
-		    parse_float(string, RECEIVE_BUFFER_SIZE, 2));
-	else if (starts_with(string, "set angular_acceleration "))
-		set_angular_acceleration(
-		    parse_float(string, RECEIVE_BUFFER_SIZE, 2));
```
Comment (Peque) [inline command.c:43]: Can you remove the pieces from `connect_bluetooth.py` too?

File: command.c [hunk 4]
```diff
@@ -30,8 +30,8 @@ static void process_command(char *string)
 	else if (starts_with(string, "set wheels_separation "))
 		set_wheels_separation(
 		    parse_float(string, RECEIVE_BUFFER_SIZE, 2));
-	else if (starts_with(string, "set max_linear_speed "))
-		set_max_linear_speed(
+	else if (starts_with(string, "set run_max_linear_speed "))
+		set_run_max_linear_speed(
```
Comment (Peque) [inline command.c]: I would call it `max_linear_speed` (I think "run" is implied, as the run speed is higher than the search speed and the search speed is limited by other factors, not a calibratable robot-dependent parameter.

**Review by Peque — CHANGES_REQUESTED** [review]

File: encoder.c (hunk 1 above)
Comment (cua-cua) [reply encoder.c]: Changed on 9d1c07f74fd5ea

File: command.c (hunk 2 above)
Comment (cua-cua) [reply command.c:66]: :stuck_out_tongue:

File: command.c (hunk 3 above)
Comment (cua-cua) [reply command.c:43]: `connect_bluetooth.py` is outside mmlib... maybe should we include the scripts on mmlin? Changed on bulebule project in any case (pull request #404)

File: command.c (hunk 4 above)
Comment (cua-cua) [reply command.c]: There is already a static variable named `max_linear_speed` on speed.c module that is different than `run_max_linear_speed` from config.c module. I f you have a better name..

File: command.c (hunk 3 above)
Comment (Peque) [reply command.c:43]: Probably in a different project. We will think about it... some day.

File: command.c (hunk 4 above)
Comment (Peque) [reply command.c]: Linear speed limit (?). Otherwise okay.

Comment (cua-cua) [reply command.c]: 9f70efa23b

**Review by Peque — APPROVED** [review]

**Pull request merged**